### ISSUE TITLE
The "id" argument must be of type string. Received undefined

### ISSUE THREAD

**Issue by AlexanderProd** [fedehub93/gatsby-source-shopify-translations#7]
I'm getting this error during the dev and build process. 

```
ERROR #11321  PLUGIN

"gatsby-source-shopify-translations" threw an error while running the onCreatePage lifecycle:

The "id" argument must be of type string. Received undefined

TypeError: The "id" argument must be of type string. Received undefined
```

I've been ignoring it so far because it didn't seem to cause any problems but now I've tried to build the page and the process fails because of this error. 

I'm wondering where this error comes from and If anybody else noticed it too.

I'm using `v0.4.0` but the error has been happening on earlier versions too.

**Comment by fedehub93** [comment]
Can you tell me both gatsby and gatsby-source-shopify version? Do you upgrade to latest version? Because there have been some schema changes and I haven't had a chance to see exactly what is changed yet.

[Release Note - https://www.gatsbyjs.com/docs/reference/release-notes/v4.11/](https://www.gatsbyjs.com/docs/reference/release-notes/v4.11/)
_All id fields that come back from the Shopify API have now been mapped to shopifyId so that id is always intrinsic to Gatsby._

**Comment by AlexanderProd** [comment]
I'm using gatsby `4.10.3` and gatsby-source-shopify `6.10.2`

**Comment by AlexanderProd** [comment]
This is what my `gatsby-node.js` file looks like: 

```JS
exports.createPages = async ({ graphql, actions }) => {
  const { createPage } = actions

  const productsQuery = await graphql(`
    {
      products: allShopifyProduct {
        edges {
          node {
            id
            title
            storefrontId
            handle
          }
        }
      }
      translatedProducts: allShopifyTranslatedProduct {
        edges {
          node {
            title
            locale
            storefrontId
            handle
          }
        }
      }
    }
  `)

  const products = productsQuery.data.products.edges
  const translatedProducts = productsQuery.data.translatedProducts.edges

  products.forEach(product => {
    const translatedProduct = translatedProducts.filter(
      tProduct => tProduct.node.storefrontId === product.node.storefrontId
    )

    translatedProduct.forEach(tProduct => {
      const newProduct = {
        ...product,
        node: {
          ...product.node,
          handle: slugify(tProduct.node.title),
        },
      }
      const { locale } = tProduct.node

      createPage({
        path:
          locale === 'de'
            ? `/shop/${slugify(tProduct.node.title)}`
            : `/${locale}/shop/${slugify(tProduct.node.title)}`,
        component: path.resolve('./src/templates/ProductPage/index.js'),
        context: {
          id: newProduct.node.id,
          locale: locale,
          title: newProduct.node.title,
          storefrontId: newProduct.node.storefrontId,
          originalPath: product.node.handle,
          handle: product.node.handle,
          isAlreadyTranslated: true,
        },
      })
    })
  })
}
```

**Comment by AlexanderProd** [comment]
the stack trace says the error happens in the `onCreatePage.js` at `31:23` in the dist. I'm wondering since in there is nothing called `id` in the whole file.

**Comment by AlexanderProd** [comment]
I've figured out the error happened because I wasn't using the `configPath` key in the plugin config because I didn't need it. (see #8 )
So `configLocales` in `onCreatePage.js` was undefined.

**Comment by fedehub93** [comment]
I've released 0.4.1 verision with your changes. 
Thank you again for contributing to this project.
